server/src/main/java/com/google/musicanalysis/cache/AnalysisCache.java

```diff
@@ -0,0 +1,87 @@
+package com.google.musicanalysis.cache;
+
+import com.google.musicanalysis.types.*;
+import java.lang.Object;
+import java.lang.ClassNotFoundException;
+import java.time.Instant;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.io.File;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.IOException;
+
+
+public class AnalysisCache {
+    private HashMap<String, AnalysisGroup> responseMap;
+    
+    public AnalysisCache(){
+        this.responseMap = new HashMap<String, AnalysisGroup>();
+    }
+
+    public void open() {
+        HashMap<String, AnalysisGroup> responseData = null;
+
+        int file = createFile(); // this will have use later.
```

Comment:
is this comment necessary? or is this to explain why file is not being used now (and will be used in a future PR)

Reply:
The second one

Reply:
Remove comments that reference future PR's unless they are explicit and well structured TODO's. Comments like this are contingent on you actually making a follow up change, and a prone to being abandoned. This means that 6 months from now, some poor developer will be reading this code and see "this will have use later." and cry because they can't contact you and ask what it means. It's easy for enterprise code to become littered with these after a year.

A properly structured TODO tells your future self explicit details of what needs to happen next so that a different developer (or even a more forgetful you) can implement the change without needing any unwritten context. Ex:
```
function sum(left: number, right: number) {
  let total = left;
  // TODO: This doesn't account for negative numbers and could be more efficient if
  // we used the `+` operator.
  while (right > 0) {
    left++;
    right--;
  }
  return total;
}
```